### PR TITLE
Exclude `buildSrc` sources in settings script model

### LINKED ISSUE #10417: Kotlin DSL settings IDE model shouldn't include buildSrc sources
This is a follow up to https://github.com/gradle/gradle/pull/10305

### Expected Behavior

Kotlin DSL IDE model for settings scripts shouldn't include `buildSrc` *sources*.

### Current Behavior

https://github.com/gradle/gradle/blob/89d4dd396f592022ce2deb9f2f187a7c5cf4b4af/subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinSettingsScriptModelCrossVersionSpec.groovy#L117

This test and the one above are now wrong, they should NOT include buildSrc *sources*. This won't be user visible in the editor because the classpath is correct. Fixing this would make the response for settings scripts smaller.

### Context

https://github.com/gradle/gradle/pull/10305

### PR DESCRIPTION
**Pull request by hegyibalint**
Fixes #10417

### CODE REVIEW

File: subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/resolver/SourcePathProviderTest.kt [hunk 1]
```diff
@@ -40,6 +40,7 @@ class SourcePathProviderTest : FolderBasedTest() {
         assertThat(
             sourcePathFor(
                 classPath = ClassPath.EMPTY,
+                scriptFile = null,
```
Comment (jbartok) [inline subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/resolver/SourcePathProviderTest.kt:43]: I think there should be a test in this class covering the recent changes.

**Review by jbartok — APPROVED** [review]
LGTM, had a single comment.

**Comment by eskatos** [comment]
@bot-gradle help

**Comment by eskatos** [comment]
@bot-gradle test ACT

File: subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinSettingsScriptModelCrossVersionSpec.groovy [hunk 2]
```diff
@@ -22,6 +22,7 @@ import org.gradle.test.fixtures.file.LeaksFileHandles
 import org.gradle.util.GradleVersion
 
 import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.not
 
 @TargetGradleVersion(">=5.4")
```
Comment (eskatos) [inline subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinSettingsScriptModelCrossVersionSpec.groovy:27]: Gradle >= 5.4 will not include your changes and behave as you expect.
You should add a new cross version test in `r75` package, duplicated from this one but asserting the new behavior.

**Review by eskatos — CHANGES_REQUESTED** [review]

File: subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinSettingsScriptModelCrossVersionSpec.groovy (hunk 2 above)
Comment (hegyibalint) [reply subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinSettingsScriptModelCrossVersionSpec.groovy]: PTAL, I've separated the test

File: subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/resolver/SourcePathProviderTest.kt (hunk 1 above)
Comment (hegyibalint) [reply subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/resolver/SourcePathProviderTest.kt:42]: Fair enough, added some extra tests focusing on the attribute.

**Comment by eskatos** [comment]
@bot-gradle test ACT

File: subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinSettingsScriptModelCrossVersionSpec.groovy (hunk 2 above)
Comment (hegyibalint) [reply subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/KotlinSettingsScriptModelCrossVersionSpec.groovy]: Are you sure about this? I find no example of this. E.g. `KotlinDslScriptsModelCrossVersionSpec` has the package name as `@TargetGradleVersion`, and does not constraint the minimal version.

Is this case here special in any way?

**Comment by hegyibalint** [comment]
@bot-gradle test ACT

**Review by eskatos — APPROVED** [review]
LGTM :+1:

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Pull request merged**